### PR TITLE
NukeOps Setup

### LINKED ISSUE #972: Set up the [Nuke Op vs Station] gamemode B: 150
------------------------------------
Task Description:
------------------------------------

The task is fairly straight forward. We need to change our current
team death match game mode to a Nuke Ops vs Station game mode.

All of the game mechanics for the Nuke ops, shuttles, station, matrix move,
networking is all done we just need you to alter the game manager code and a few
other things to meet the criteria below.

We also need to set up a way point system for our escape shuttle to arrive
at a position on the map after a certain amount of time has elapsed.

Some Notes:
------------------------------------

Fork our repo to see the current game mode:
https://github.com/unitystation/unitystation
(Use 2018.2 and run the Lobby scene, you can even build a player
by including the lobby, outpostdm and flashlightdm scenes)

Criteria:
------------------------------------

- [ ] Remove all of the old reporting code (the stat reports from the gamemanager to the chat feed) from the TDM game state and now implement code to determine who won the end of the Nuke Ops vs Station round.

----
#### Rules to determine which side wins:

- [ ] If all nuke ops died and 1 station member is alive and on the shuttle after the 10 minute round time ends then the Station crew wins

- [ ] If no one is on the shuttle after 10 minutes then the Nuke Ops win (you will need to develop some logic to check the shuttles children for players who are still alive on the server. Players move to the tree of the shuttle or station they are on)

- [ ] If the nuke is set off on the station then the Nuke Ops win

----- 
- [ ] Print which side won to the chat feed and wait a minute after the round ends before playing next round

- [ ] Set the timer to count down from 10 minutes and then set the escape shuttle to arrive at the station at the end of that timer (You will need to do some trial and error here to get it arriving at the time it is meant to). Count down again for 1 minutes and then depart the shuttle. End the round after about a minute of flight time and count all of the survivors that made it onto the escape shuttle.

- [ ] Apply all of the criteria to the OutpostDeathmatch scene. Try to also apply the changes to the FlashlightDeathmatch scene (we know there is obstacles to overcome on the flashlight scene so if it becomes too tricky then a partial implementation will be good enough, just as long as it works perfectly on OutpostDM).

- [ ] Test all of the criteria in a multiplayer setting by creating two builds that include the lobby, outpostdm and flashlightdm scenes. Play with 2 players for a few rounds to confirm

Some Notes:
------------------------------------

- Shuttle movement is controlled by the MatrixMove.cs component. Create a new component for the Escape shuttle that can use the Matrix Move to accomplish the waypoint movement and the automatic start so that it arrives at the station on time.

- Check out GameManager.cs to locate some of the current code

- Check out PlayerList.cs for helpful references and tools to do with the current players

- Nuke ops can blow the nuke by using the code that they get on round start and entering it into the nuke near the middle of the outpost station (note: there is no nuke on flashlight at the moment) 

Bounty of 150usd will be paid to the contributor on a successful merge

### PR DESCRIPTION
**Pull request by dapocalypse**
### Purpose
Fixes #972 

NukeOps is not set up. This PR sets up the nuke ops gamemode.

### Open Questions and Pre-Merge TODOs

- [x]  This fix is tested on the same branch it is PR'ed to.
- [x]  I correctly commented my code
- [x]  My code is indented with tabs and not spaces
- [x]  This PR does not include any unnecessary .meta, .prefab or .unity (scene) changes
- [x]  This PR does not bring up any new compile errors
- [x]  This PR has been tested in editor
- [x]  This PR has been tested in multiplayer

### Notes:
Paypal: dedhiadev5@gmail.com
B: 100

### CODE REVIEW

**Comment by dapocalypse** [comment]
This is WIP until I say otherwise.

File: UnityProject/Assets/Scripts/Editor/Tools/SearchandDestroy.cs [hunk 1]
```diff
@@ -143,7 +143,7 @@ private void OnGUI()
 								DestroyImmediate(component, true);
 								//Logger.Log
 								//			PrefabUtility.ReplacePrefab(castGO, castPrefab, ReplacePrefabOptions.Default);
-								PrefabUtility.ReplacePrefab(cast, PrefabUtility.GetCorrespondingObjectFromSource(cast), ReplacePrefabOptions.ConnectToPrefab);
+								// PrefabUtility.ReplacePrefab(cast, PrefabUtility.GetCorrespondingObjectFromSource(cast), ReplacePrefabOptions.ConnectToPrefab);
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/Editor/Tools/SearchandDestroy.cs:146]: why?

File: UnityProject/Assets/Scripts/Food/FoodBehaviour.cs [hunk 2]
```diff
@@ -7,8 +7,6 @@
 /// </summary>
 public class FoodBehaviour : NetworkBehaviour
 {
-	//TODO remove after deathmatches
-	[Header("Being used for TDM")] public int healAmount;
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/Food/FoodBehaviour.cs:11]: So food no longer heals? We don't have any medical supplies in yet

File: UnityProject/Assets/Scripts/Health/Living/PlayerHealth.cs [hunk 3]
```diff
@@ -219,54 +219,7 @@ protected override void OnDeathActions()
 				PlayerMove pm = gameObject.GetComponent<PlayerMove>();
 
 				ConnectedPlayer player = PlayerList.Instance.Get(gameObject);
-				
-				string killerName = "Stressful work";
-				if (LastDamagedBy != null)
-				{
-					killerName = PlayerList.Instance.Get(LastDamagedBy).Name;
-				}
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/Health/Living/PlayerHealth.cs:227]: We should leave the killer name stuff, we still want to know who killed who

File: UnityProject/Assets/Scripts/Health/Living/PlayerHealth.cs [hunk 4]
```diff
@@ -219,54 +219,7 @@ protected override void OnDeathActions()
...
-							SpawnPoint.GetJobDepartment(LastDamagedBy.GetComponent<PlayerScript>().JobType);
-						JobDepartment victimDepartment =
-							SpawnPoint.GetJobDepartment(gameObject.GetComponent<PlayerScript>().JobType);
-
-						if (killerDepartment == victimDepartment)
-						{
-							departmentKillText = ", losing " + killerDepartment.GetDescription() +
-							                     " 1 point for team killing!";
-						}
-						else
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/Health/Living/PlayerHealth.cs:257]: There doesn't seem to be any attempt here to attribute a kill to either NukeOps or the Station crew. Instead you have just removed all the reporting code

File: UnityProject/Assets/Scripts/PlayGroups/PlayerNetworkActions.cs [hunk 5]
```diff
@@ -625,14 +584,6 @@ public void CmdEatFood(GameObject food, string fromSlot)
...
 		FoodBehaviour baseFood = food.GetComponent<FoodBehaviour>();
 		soundNetworkActions.CmdPlaySoundAtPlayerPos("EatFood");
-		PlayerHealth playerHealth = GetComponent<PlayerHealth>();
-
-		//FIXME: remove health and blood changes after TDM
-		//and use this Cmd for healing hunger and applying 
-		//food related attributes instead:
-		playerHealth.AddHealth(baseFood.healAmount);
-		playerHealth.BloodLevel += baseFood.healAmount;
-		playerHealth.StopBleeding();
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/PlayGroups/PlayerNetworkActions.cs:635]: There was no requirement to remove this

File: UnityProject/Assets/Scripts/Shuttles/MatrixMove.cs [hunk 6]
```diff
@@ -523,6 +617,6 @@ public void RotateTo( Orientation desiredOrientation )
 #endif
 }
 [Serializable]
-public class OrientationEvent : UnityEvent<Orientation,Orientation> {}
+public class OrientationEvent : UnityEvent<Orientation, Orientation> { }
 [Serializable]
-public class DualFloatEvent : UnityEvent<float,float> {}
\ No newline at end of file
+public class DualFloatEvent : UnityEvent<float, float> { }
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/Shuttles/MatrixMove.cs:622]: Oh dear god. I'm not reading this diff. Unsure of what you have changed here.

File: UnityProject/Assets/Scripts/Managers/MatrixManager.cs [hunk 7]
```diff
@@ -11,341 +11,356 @@
...
+            netId = componentInParent.netId;
+        }
+
+        if (netId == NetworkInstanceId.Invalid)
+        {
+            Logger.LogWarning($"Invalid NetID for matrix {matrix.gameObject.name}!", Category.Matrix);
+        }
+
+        return netId;
+    }
```
Comment (DooblyNoobly) [inline UnityProject/Assets/Scripts/Managers/MatrixManager.cs:365]: What was changed here? I'm not reading through the diff to find out as it was reformatted

**Review by DooblyNoobly — CHANGES_REQUESTED** [review]
Needs a lot of work. There is no EscapeShuttle component that controls the movement of the shuttle serverside to escape. There is also a lot of unnecessary changes in this PR. You are getting better but you still need a bit more experience. I would suggest tackling a simplier problem first before attempting something as complex as trying to automate EscapeShuttle arrival.

A lot of files were included in this PR that didn't need to be there:
UnityProject/ProjectSettings/UnityConnectSettings.asset
UnityProject/Assets/Plugins/Facepunch.Steamworks/Facepunch.Steamworks.dll.mdb
UnityProject/Assets/Scenes/Lobby.unity
UnityProject/Assets/Scenes/OutpostDeathmatch.unity
UnityProject/Assets/Scripts/Editor/Tools/SearchandDestroy.cs
UnityProject/Assets/Scripts/Managers/New Prefab.prefab
UnityProject/Assets/Scripts/Managers/New Prefab.prefab.meta

File: UnityProject/Assets/Scripts/Editor/Tools/SearchandDestroy.cs (hunk 1 above)
Comment (dapocalypse) [reply UnityProject/Assets/Scripts/Editor/Tools/SearchandDestroy.cs:146]: This caused an error.

**Pull request closed without merging**